lib/src/screenshots.dart

```diff
@@ -0,0 +1,327 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:pana/pana.dart';
+import 'package:pana/src/utils.dart';
+import 'package:path/path.dart' as path;
+import 'package:pubspec_parse/pubspec_parse.dart' as p;
+
+final maxFileSizeInBytes = 2097152; // 2MB
+final maxFileSizeInMegaBytes = maxFileSizeInBytes / (1024 * 1024);
+final maxNumberOfScreenshots = 5;
+
+/// [ScreenshotResult] holds either a [ProcessedScreenshot] or a list of
+/// problems describing what went wrong when trying to generate thumbnails and
+/// screenshot.
+class ScreenshotResult {
+  // The temporary directory where the processed screenshots are stored.
+  final String? _outDir;
+  ProcessedScreenshot? processedScreenshot;
+  List<String> problems;
+
+  ScreenshotResult._(this.processedScreenshot, this.problems, this._outDir);
+
+  factory ScreenshotResult.success(
+      ProcessedScreenshot processedScreenshot, String outDir) {
+    return ScreenshotResult._(processedScreenshot, [], outDir);
+  }
+
+  factory ScreenshotResult.failed(List<String> problems) {
+    return ScreenshotResult._(null, problems, null);
+  }
+
+  /// The path to where the generated WebP thumbnail is temporarily stored.
+  /// Should only be used if [processedScreenshot] is not null.
+  String get webpThumbnailPath =>
+      path.join(_outDir!, processedScreenshot!.webpThumbnail);
```

Comment:
Why not make this a `Future<Uint8List> readWebpThumbnail() async` method?

The actual path within `_outDIr` is an implementation detail. The only thing the public pana interface is that `storeResource` is called with the path that appears in `processedScreenshot`.

Reply:
Or we should make this a property `final Uint8List webpThumbnail;` that way we won't need to manage the life-cycle of the `_outDir` which is currently leaked?